gempy/core/data.py

```diff
@@ -1975,6 +1975,116 @@ def create_orientation_from_surface_points(self, surface_points: SurfacePoints,
 
         return np.array([*center, *orientation, *normal])
 
+    def create_orientation_from_nn(self, surface_points,  searchcrit):
+        """
+        Calculates the orientation from neighbour points of the same surface
+        by given radius (radius-search) or fix number (knn).
+    
+        Parameters
+        ----------
+        self : geo_model
+            GemPy-model.
+        surface_points: Pandas-dataframe
+            Contains the dataframe of the (point-)data from the GemPy-model.
+        searchcrit : int or float
+            if is int: uses knn-search.
+            if is float: uses radius-search.
+        """
+        
+        from sklearn.neighbors import NearestNeighbors
+        from sklearn.preprocessing import normalize
+    
+        # extract surface names
+        surfaces = np.unique(surface_points['surface'])
+        nbrs = []
+        search_id = []
+        # for each surface
+        if isinstance(searchcrit, int):  # in case knn-search
```

Comment:
Once you have the previous function that allows to create one orientation from 1 surface point you could add another function (in the same module) such as

`set_all_orientations_from_surface_points` where you loop all the surface points as you are doing down here.

Splitting into this 3 functions would help a lot to have a clear picture of what is happening and avoid repeating code too much. Also it makes easier
to test each function separetly.